README.md

```diff
@@ -71,39 +268,210 @@ See [Setting up debug environment](doc/debugging.md) for more details.
+
+iOS devices(MacOS only):
+
+```
+xcrun simctl list --json devices
+```
+
+Android devices:
+
+```
+adb devices
+```
+
+**NOTE:** If you want run application on iOS devices make sure you have `ios-deploy` installed globally.
+
+```npm install -g ios-deploy```
+
+## Setting up the react-native packager
+
+To use a custom port for the `react-native` packager:
+
+```json
+{
+  "react-native": {
+    "packager" : {
+      "port": portNumber
+    }
+  }
+}
+```
+
+If you change this port, then for iOS device and simulator scenarios you will have to modify the native code files. Instruction [here](https://blog.binoy.io/running-react-native-on-a-different-port-7deb43887cd4).
+
+If you use android, you need to change debug server by:
+
+1. CTRL+M(CMD+M) in the emulator
+2. Go to Dev Settings
+3. Debug server host for device => enter ‘localhost:\<yourPortNumber\>’.
+4. Reload application (double R)
+
+**NOTE:** Some aspects of React Native hard-code the port to the default as specified in [this issue](https://github.com/facebook/react-native/issues/9145).
+
+
+## Change project root
+
+To specify a subfolder in which the react-native project is located, set `react-native-tools.projectRoot`. You can use either an absolute or relative path here:
+
+```json
+{
+  "react-native-tools": {
+    "projectRoot": "./your/react-native/project"
+  }
+}
+```
 
-The **Packager** commands allow you to start/stop the [**Metro Bundler**](https://github.com/facebook/metro-bundler) (formerly React Packager).
+# Developing inside a Docker Container
+
+The extension supports [VS Code Remote Development](https://code.visualstudio.com/docs/remote/remote-overview) features on Linux. Please follow the [VS Code official documentation](https://code.visualstudio.com/docs/remote/containers) to setup your environment to use remote development approach.
+
+You can use [official React Native Docker image](https://hub.docker.com/r/reactnativecommunity/react-native-android) provided by [react-native-community](https://github.com/react-native-community/docker-android).
+
+Here are the steps to run React Native debugging inside Docker Container on a real Android device:
+
+1. Open Command Palette and run the following command
+    ```
+    Remote-Containers: Add Development Container Configuration Files...
+    ```
+    Then select `Existing Dockerfile` to create `.devcontainer/devcontainer.json` configuration file.
+1. Сreate Dockerfile extending [reactnativecommunity/react-native-android image](https://hub.docker.com/r/reactnativecommunity/react-native-android). For example you can use the following Dockerfile:
+    ```
+    FROM reactnativecommunity/react-native-android:latest
+
+    RUN npm install -g expo-cli react-native-cli
+    ```
 
-## Using Expo
+1. Configure your `devcontainer.json` file just about like this: <br> **NOTE**: This is just a sample of configuration, you can modify your `devcontainer.json` file as you need.
```

Comment:
1. Configure your `devcontainer.json` file as needed. Below is an sample configuration: